test/libsolidity/SemanticTest.cpp

```diff
@@ -58,49 +59,51 @@ SemanticTest::SemanticTest(
 	m_sources(m_reader.sources()),
 	m_lineOffset(m_reader.lineNumber()),
 	m_enforceViaYul(_enforceViaYul),
+	m_enforceCompileToEwasm(_enforceCompileToEwasm),
 	m_enforceGasCost(_enforceGasCost),
-	m_enforceGasCostMinValue(_enforceGasCostMinValue)
+	m_enforceGasCostMinValue(std::move(_enforceGasCostMinValue))
 {
-	string choice = m_reader.stringSetting("compileViaYul", "default");
-	if (choice == "also")
+	string compileViaYul = m_reader.stringSetting("compileViaYul", "default");
+	if (compileViaYul == "also")
 	{
-		m_runWithYul = true;
-		m_runWithoutYul = true;
+		m_testCaseWantsYulRun = true;
+		m_testCaseWantsLegacyRun = true;
 	}
-	else if (choice == "true")
+	else if (compileViaYul == "true")
 	{
-		m_runWithYul = true;
-		m_runWithoutYul = false;
+		m_testCaseWantsYulRun = true;
+		m_testCaseWantsLegacyRun = false;
 	}
-	else if (choice == "false")
+	else if (compileViaYul == "false")
 	{
-		m_runWithYul = false;
-		m_runWithoutYul = true;
-		// Do not try to run via yul if explicitly denied.
+		m_testCaseWantsYulRun = false;
+		m_testCaseWantsLegacyRun = true;
+		// Do not try to run via yul and ewasm if explicitly denied.
 		m_enforceViaYul = false;
+		m_enforceCompileToEwasm = false;
 	}
-	else if (choice == "default")
+	else if (compileViaYul == "default")
 	{
-		m_runWithYul = false;
-		m_runWithoutYul = true;
+		m_testCaseWantsYulRun = false;
+		m_testCaseWantsLegacyRun = true;
 	}
 	else
-		BOOST_THROW_EXCEPTION(runtime_error("Invalid compileViaYul value: " + choice + "."));
+		BOOST_THROW_EXCEPTION(runtime_error("Invalid compileViaYul value: " + compileViaYul + "."));
```

Comment:
This could now be shortened to something like:
```
m_testCaseWantsYulRun = (compileViaYul == "also" || compileViaYul == "true");
m_testCaseWantsLegacyRun = (compileViaYul == "also" || compileViaYul == "false" || compileViaYul == "default");
if (compileViaYul != "also" && compileViaYul != "ftrue" && compileViaYul != "false" && compileViaYul != "default")
    BOOST_THROW_EXCEPTION(runtime_error("Invalid compileViaYul value: " + compileViaYul + "."));
```
or
```c++
set<string> const compileViaYulAllowedValues = {"also", "true", "false", "default"};
set<string> const yulRunTriggers = {"also", "true"};
set<string> const legacyRunTriggers = {"also", "false", "default"};

if (compileViaYulAllowedValues.count(compileViaYul) == 0)
    BOOST_THROW_EXCEPTION(runtime_error("Invalid compileViaYul value: " + compileViaYul + "."));

m_testCaseWantsYulRun = (yulRunTriggers.count(compileViaYul) > 0);
m_testCaseWantsLegacyRun = (legacyRunTriggers.count(compileViaYul) > 0);
```

Reply:
Looks like we have a `contains()` helper in `CommonData.h`. With this, it could be even better:
```c++
if (!contains({"also", "true", "false", "default"}, compileViaYul))
    BOOST_THROW_EXCEPTION(runtime_error("Invalid compileViaYul value: " + compileViaYul + "."));

m_testCaseWantsYulRun = contains({"also", "true"}, compileViaYul);
m_testCaseWantsLegacyRun = contains({"also", "false", "default"}, compileViaYul);
```